pilot/pkg/security/authz/model/model.go

```diff
@@ -265,3 +272,42 @@ func (m *Model) Generate(service *ServiceMetadata, forTCPFilter bool) *envoy_rba
 	}
 	return policy
 }
+
+// getPrincipalsIncludingAliases returns a list of users encoded in SPIFFE format with the local
+// trust domain and its aliases.
+// For example, for a user "bar" in namespace "foo".
+// If the local trust domain is "td2" and its alias is "td1" (migrating from td1 to td2),
+// getPrincipalsIncludingAliases returns ["td2/ns/foo/sa/bar", "td2/ns/foo/sa/bar]].
+func getPrincipalsIncludingAliases(trustDomain string, trustDomainAliases []string, principals []string) []string {
+	// If trust domain aliases are empty, return the existing principals.
+	if len(trustDomainAliases) == 0 {
+		return principals
+	}
+	principalsIncludingAliases := []string{}
+	for _, principal := range principals {
+		trustDomainFromPrincipal := extractTrustDomainFromAuthzPrincipal(principal)
+		if trustDomainFromPrincipal == "" {
+			return principals
```

Comment:
This means if an invlid principal will cause all the principals to be skipped for alias replacing, is this expected?

Reply:
Yes, if there is an invalid principal, I think we should just return whatever exists before. This should not happen though, since the users should know if their RBAC rules work or not in their existing systems (and migration happens later).

Reply:
What if the principal is `*` or  `*/ns/foo/sa/sleep`?

Reply:
For "*" it should already work because it will return the existing principals, which doesn't care about trust domain. For prefix/suffix, I added a TODO and will implement it in the next PR. This one is pretty big already.